src/components/scene/device-motion-permission-ui.js

```diff
@@ -0,0 +1,161 @@
+/* global DeviceOrientationEvent  */
+var registerComponent = require('../../core/component').registerComponent;
+var utils = require('../../utils/');
+var bind = utils.bind;
+
+var constants = require('../../constants/');
+
+var DEVICE_PERMISSION_FULL_CLASS = 'a-device-motion-permission-full';
+var DEVICE_PERMISSION_FULL_CENTER_CLASS = 'a-device-motion-permission-full-center';
+var DEVICE_PERMISSION_ACCEPT_CLASS = 'a-device-motion-permission-accept';
+var DEVICE_PERMISSION_CANCEL_CLASS = 'a-device-motion-permission-cancel';
+var DEVICE_PERMISSION_REQUEST_CLASS = 'a-device-permision-request';
+
+/**
+ * UI for enabling device motion permission
+ */
+module.exports.Component = registerComponent('device-motion-permission-ui', {
+  schema: {
+    enabled: { default: true },
+    orientationChange: {default: null},
+    deviceOrientation: {default: null},
+    deviceMotionEl: { default: '' }
+  },
+  init: function () {
+    this.deviceMotionEl = null;
+    this.onDeviceMotionClick = bind(this.onDeviceMotionClick, this);
+    this.onOrientationChangeClick = bind(this.onOrientationChangeClick, this);
+    this.grantedDeviceMotion = bind(this.grantedDeviceMotion, this);
+    if (typeof window.orientation !== 'undefined') {
```

Comment:
This if statement can be simplified to:
```
if (DeviceOrientationEvent && DeviceOrientationEvent.requestPermission) {
      this.deviceMotionEl = createDeviceMotionPermissionDialog(this.onDeviceMotionClick, this);
      this.el.appendChild(this.deviceMotionEl);
}
```
See [suggested edits](https://github.com/dmarcos/aframe/blob/b4be06e206821cedaa37a6b4512f4bd6641be1bd/src/components/scene/device-motion-permission-ui.js#L27)

Reply:
The reasons for the structure here follow. I have tried these in different browsers and devices, including multiple iOS versions. There are other ways to do this, but going too simple or too complex will lead to similar outcomes. Thoughts?

iOS13
 - DeviceOrientationEvent.requestPermission needs to be called, not only tested to exist;
iOS and Android
 - grantedDeviceMotion is called by mobile devices (window.orientation)
Desktop Browsers
 - Do not call grantedDeviceMotion

Reply:
DeviceOrientationEvent.requestPermission is already called when the user clicks on the allow button:
https://github.com/aframevr/aframe/pull/4303/files/8c30aaf2ce8b732dcc8b1644322fcad5f08e7036#diff-81b7296fca054a84c9266fd8fdebee05R75

Do we need to call it twice? 

lf think if `DeviceOrientationEvent.requestPermission` is not available we don't need to call any additional methods or initialize the dialog.